### PR TITLE
update customers with testcase

### PR DESCRIPTION
**Pull request by ankitdas13**
Api reference for `all` : https://razorpay.com/docs/api/customers/#fetch-all-customers

### CODE REVIEW

File: documents/customers.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| name*          | string      | Name of the customer                        |
+| email        | string      | Email of the customer                       |
+| contact      | string      | Contact number of the customer              |
+| notes         | object      | A key-value pair                            |
```
Comment (Madeyedexter) [inline documents/customers.md:28]: Please add `fail_existing` parameter as well.

File: documents/customers.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+}
+```
+
+-------------------------------------------------------------------------------------------------------
+
+### Fetch a customer
+```go
+customerId := "cust_1Aa00000000003"
+
+body, err := client.Cusstomer.Fetch(customerId, nil)
```
Comment (Madeyedexter) [inline documents/customers.md:133]:
```suggestion
body, err := client.Customer.Fetch(customerId, nil)
```

File: documents/customers.md (hunk 1 above)
Comment (ankitdas13) [reply documents/customers.md:29]: i added fail_existing param

File: documents/customers.md (hunk 2 above)
Comment (ankitdas13) [reply documents/customers.md]: have made spelling correction

File: resources/customer_test.go [hunk 3]
```diff
@@ -46,3 +46,12 @@ func TestCustomerEdit(t *testing.T) {
 	assert.Equal(t, err, nil)
 	utils.TestResponse(jsonByteArray, []byte(fixture), t)
 }
+
+func TestCustomerAll(t *testing.T) {
+	teardown, fixture := utils.StartMockServer(constants.CUSTOMER_URL, "customers_collection")
+	defer teardown()
+	body, err := utils.Client.Customer.All(nil, nil)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply resources/customer_test.go]: have added count parameter object

[the remark this answers is not in the document]
Comment (ankitdas13) [reply resources/customer_test.go]: also added url parse in get method

[the remark this answers is not in the document]
Comment (ankitdas13) [reply resources/customer_test.go]: also added url parse in get method

File: testdata/customers_collection.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        "entity":"customer",
+        "name":"Asa Fox",
+        "email":"asa.fox@example.com",
+        "contact":"9123456789",
+        "gstin":null,
+        "notes":[],
+        "created_at ":1234567890
+      }
+   ]
+}
```
Comment (Madeyedexter) [inline testdata/customers_collection.json:29]: ensure End of Line character at the end of json files

**Review by Madeyedexter — APPROVED** [review]

**Pull request merged**